test/unit/argument_parser/argument_parser_design_error_test.cpp

```diff
@@ -86,6 +86,33 @@ TEST(parse_test, parser_design_error)
     parser7.add_positional_option(vec, "oh oh list not at the end.");
     parser7.add_positional_option(option_value, "desc.");
     EXPECT_THROW(parser7.parse(), parser_design_error);
+
+    // using h, help, advanced-help, and export-help
+    argument_parser parser8("test_parser", 1, argv);
+    EXPECT_THROW(parser8.add_option(option_value, 'h', "", "-h is bad."),
+                 parser_design_error);
+    EXPECT_THROW(parser8.add_option(option_value, '\0', "help", "help is bad."),
+                 parser_design_error);
+    EXPECT_THROW(parser8.add_option(option_value, '\0', "advanced-help",
+                 "advanced-help is bad"), parser_design_error);
+    EXPECT_THROW(parser8.add_option(option_value, '\0', "export-help",
+                 "export-help is bad"), parser_design_error);
+
+    // using one-letter long identifiers.
+    argument_parser parser9("test_parser", 1, argv);
+    EXPECT_THROW(parser9.add_option(option_value, 'y', "z", "long identifier is one letter"),
+                 parser_design_error);
+    EXPECT_THROW(parser9.add_flag(flag_value, 'y', "z", "long identifier is one letter"),
+                 parser_design_error);
+
+    // using non-printable characters
+    argument_parser parser10("test_parser", 1, argv);
+    EXPECT_THROW(parser10.add_option(option_value, '\t', "no\n", "tab and newline don't work!"),
+                 parser_design_error);
+    EXPECT_THROW(parser10.add_flag(flag_value, '\t', "no\n", "tab and newline don't work!"),
```

Comment:
```suggestion
    EXPECT_THROW(parser10.add_flag(flag_value, 'i', "no\n", "tab and newline don't work in long ids either!"),
```
Otherwise the short id is always checked first, the exception is thrown and you are never testing the long id